PocketKit/Sources/PocketKit/Tags/PocketAddTagsViewModel.swift

```diff
@@ -1,30 +1,66 @@
 import Combine
 import Sync
 import Textile
+import Foundation
+import SharedPocketKit
 
 class PocketAddTagsViewModel: AddTagsViewModel {
     private let item: SavedItem
     private let source: Source
     private let saveAction: () -> Void
+    private var userInputListener: AnyCancellable?
+
+    var sectionTitle: TagSectionType = .allTags
 
     @Published
     var tags: [String] = []
 
+    @Published
+    var newTagInput: String = ""
+
+    @Published
+    var otherTags: [String] = []
+
     init(item: SavedItem, source: Source, saveAction: @escaping () -> Void) {
         self.item = item
         self.source = source
         self.saveAction = saveAction
 
         tags = item.tags?.compactMap { ($0 as? Tag)?.name } ?? []
+        allOtherTags()
+
+        userInputListener = $newTagInput
+            .debounce(for: .seconds(0.5), scheduler: DispatchQueue.main)
+            .sink(receiveValue: { [weak self] text in
+                self?.filterTags(with: text)
+            })
     }
 
     func addTags() {
         source.addTags(item: item, tags: tags)
         saveAction()
     }
 
-    func allOtherTags() -> [String]? {
+    func allOtherTags() {
```

Comment:
Can we doc block this?

Reply:
will do! I'll also add it to the preexisting functions too